include/deal.II/grid/tria.h

```diff
@@ -1909,20 +1886,84 @@ class Triangulation : public Subscriptor
 
   /**
    * A structure that has boost::signal objects for a number of actions that a
-   * triangulation can do to itself. See the general documentation of the
-   * Triangulation class for more information and for documentation of the
-   * semantics of the member functions.
+   * triangulation can do to itself. Please refer to the
+   * "Getting notice when a triangulation changes" section in the general
+   * documentation of the @ref Triangulation class for more information
+   * and examples.
    *
    * For documentation on signals, see
    * http://www.boost.org/doc/libs/release/libs/signals2 .
    */
   struct Signals
   {
+    /**
+     * This signal is triggered whenever the
+     * Triangulation::create_triangulation or Triangulation::copy_triangulation
+     * is called. This signal is also triggered when loading a triangulation from an
+     * archive via Triangulation::load.
+     */
     boost::signals2::signal<void ()> create;
+
+    /**
+     * This signal is triggered at the beginning of execution of
+     * the Triangulation::execute_coarsening_and_refinement function (which is
+     * itself called by other functions such as Triangulation::refine_global). At
+     * the time this signal is triggered, the triangulation is still unchanged.
+     */
     boost::signals2::signal<void ()> pre_refinement;
+
+    /**
+     * This signal is triggered at the end of execution of the
+     * Triangulation::execute_coarsening_and_refinement function when the
+     * triangulation has reached its final state
+     */
     boost::signals2::signal<void ()> post_refinement;
-    boost::signals2::signal<void (const Triangulation<dim, spacedim> &original_tria)> copy;
+
+    /**
+     * This signal is triggered for each cell that is going to be coarsened.
+     *
+     * @note the signal parameter @p cell is correspond to the upcoming cell
+     * after coarsening rather than a currently active cell.
+     * In another word, all children of @p cell will be deleted after coarsening
+     * and @p cell will become the new active cell then.
+     */
+    boost::signals2::signal<void (const Triangulation<dim, spacedim>::cell_iterator &cell)> pre_coarsening_on_cell;
+
+    /**
+     * This signal is triggered for each cell that just has been refined.
+     *
+     * @note the signal parameter @p cell is correspond cell that just has been
+     * refined rather than a resulting children cell. When this signal is triggered,
+     * the @p cell is no longer active.
```

Comment:
Same here -- are you saying that it's the parent cell? If so, maybe rewrite the same way as above.

(Grammar hints: (i) You need to start the sentence with a capital letter in "the". (ii) The correct way to use the word is not "is correspond to" but "corresponds to".)